optimum/deepsparse/modeling_diffusion.py

```diff
@@ -0,0 +1,527 @@
+                The ONNX Runtime inference session associated to the VAE decoder.
+            text_encoder_session (`ort.InferenceSession`):
+                The ONNX Runtime inference session associated to the text encoder.
+            unet_session (`ort.InferenceSession`):
+                The ONNX Runtime inference session associated to the U-NET.
+            config (`Dict[str, Any]`):
+                A config dictionary from which the model components will be instantiated. Make sure to only load
+                configuration files of compatible classes.
+            tokenizer (`CLIPTokenizer`):
+                Tokenizer of class
+                [CLIPTokenizer](https://huggingface.co/docs/transformers/v4.21.0/en/model_doc/clip#transformers.CLIPTokenizer).
+            scheduler (`Union[DDIMScheduler, PNDMScheduler, LMSDiscreteScheduler]`):
+                A scheduler to be used in combination with the U-NET component to denoise the encoded image latents.
+            feature_extractor (`Optional[CLIPFeatureExtractor]`, defaults to `None`):
+                A model extracting features from generated images to be used as inputs for the `safety_checker`
+            vae_encoder_session (`Optional[ort.InferenceSession]`, defaults to `None`):
+                The ONNX Runtime inference session associated to the VAE encoder.
+            model_save_dir (`Optional[str]`, defaults to `None`):
+                The directory under which the model exported to ONNX was saved.
+        """
+
+        self._internal_dict = config
+        self.model_save_dir = model_save_dir
+
+        self.is_static = static_shapes
+
+        self.vae_decoder_path = vae_decoder_path
+        self.vae_encoder_path = vae_encoder_path
+        self.text_encoder_path = text_encoder_path
+        self.unet_path = unet_path
+        self.text_encoder_2_path = text_encoder_2_path
+
+        self._device = device.upper()
+
+        self.compile(height=None, width=None)
+
+        self.tokenizer = tokenizer
+        self.tokenizer_2 = tokenizer_2
+        self.scheduler = scheduler
+        self.feature_extractor = feature_extractor
+        self.safety_checker = None
+
+        sub_models = {
+            DIFFUSION_MODEL_TEXT_ENCODER_SUBFOLDER: self.text_encoder,
+            DIFFUSION_MODEL_UNET_SUBFOLDER: self.unet,
+            DIFFUSION_MODEL_VAE_DECODER_SUBFOLDER: self.vae_decoder,
+            DIFFUSION_MODEL_VAE_ENCODER_SUBFOLDER: self.vae_encoder,
+            DIFFUSION_MODEL_TEXT_ENCODER_2_SUBFOLDER: self.text_encoder_2,
+        }
+
+        # Modify config to keep the resulting model compatible with diffusers pipelines
+        for name in sub_models.keys():
+            self._internal_dict[name] = (
+                ("diffusers", "OnnxRuntimeModel") if sub_models[name] is not None else (None, None)
+            )
+        self._internal_dict.pop("vae", None)
+
+    def compile(self, height, width, batch_size=1):
+        unet_batch_size = 2
```

Comment:
is there a batch size to set for the whole pipeline? how does this interact with the internal batch size - just a simple multiplication?

Reply:
The whole pipeline works with a default batch size of 1. What do you mean by `internal batch size` ?